src/app/components/my-stats/my-stats.component.ts

```diff
@@ -0,0 +1,115 @@
+
+    ) {
+        this.authenticationService.currentUser.subscribe(user => this.user = user)
+    }
+
+    ngOnInit(): void {
+        this.categoryService.getCategories().subscribe((categories) => {
+            this.pCategories = categories.filter(c => c.parent == null)
+
+            this.userStatsService.getUserStats().subscribe(stats => {
+                this.stats = stats
+                this.challengesDone = stats.challenge_stats.challenges_completed
+                this.goalsDone = stats.goal_stats.goals_completed
+
+                for(let i = 0; i<this.pCategories.length; i++){
+                    if(this.pCategories[i].pk === 64){
+                        this.basicsNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk ).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    } else if(this.pCategories[i].pk === 72)
+                        this.predefNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 83)
+                        this.conditionalsNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 91)
+                        this.loopsNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 100)
+                        this.methodsNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 107)
+                        this.arraysNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                    else if(this.pCategories[i].pk === 115)
+                        this.oopNum = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === this.pCategories[i].pk).reduce((sum, obj) => {
+                            return sum + obj.questions_solved
+                        },0)
+                }
```

Comment:
Here you are still doing the same thing. Those numbers `64`, `72`, ... is specific to your local setup. It will be different in production and in other people's computers.

You need to find the top-level categories, something like
```
this.topLevelCategories = this.pCategories.filter(category => !category.parent)
```

Then you have the categories (basics, loops, ...). You should not hard-code the categories.

then you can filter the category_stats for those categoreis.
```
for (category in this.topLevelCategories) {
  this.questionsSolved[category.pk] = this.stats.category_stats.filter(stats => stats.difficulty === 'ALL' && stats.category === category.pk).reduce((sum, obj) => {
                            return sum + obj.questions_solved
                        },0)
}
```

Then in the html you can make a for loop over `this.questionsSolved` and show the stats for all the categories.